apiserver/modelmanager/modelmanager.go

```diff
@@ -584,7 +577,35 @@ func (m *ModelManagerAPI) ModelInfo(args params.Entities) (params.ModelInfoResul
 	return results, nil
 }
 
-func (m *ModelManagerAPI) getModelInfo(tag names.ModelTag) (params.ModelInfo, error) {
+func (m *ModelManagerAPI) getCAASModelInfo(tag names.ModelTag) (params.CAASModelInfo, error) {
+	st, err := m.state.ForCAASModel(tag)
+	if errors.IsNotFound(err) {
+		return params.CAASModelInfo{}, errors.Trace(common.ErrPerm)
+	} else if err != nil {
+		return params.CAASModelInfo{}, errors.Trace(err)
+	}
+	defer st.Close()
+
+	model, err := st.CAASModel()
```

Comment:
Instead of getting the `CAASState` using `ForCAASModel` and then getting the `CAASModel` from that can't you just use the `State.GetCAASModel` method you add in this PR? That would be more efficient.

The controller UUID can be obtained from `m.state.ControllerUUID()`.